### ISSUE TITLE
Latest cargo warns about auto discovery

### ISSUE THREAD

**Issue by csmoe** [rust-lang/rust#50445]
```
warning: An explicit [[test]] section is specified in Cargo.toml which currently
disables Cargo from automatically inferring other test targets.
This inference behavior will change in the Rust 2018 edition and the following
files will be included as a test target:

* /rust/src/liballoc/tests/vec.rs
* /rust/src/liballoc/tests/string.rs
* /rust/src/liballoc/tests/cow_str.rs
* /rust/src/liballoc/tests/heap.rs
* /rust/src/liballoc/tests/binary_heap.rs
* /rust/src/liballoc/tests/str.rs
* /rust/src/liballoc/tests/linked_list.rs
* /rust/src/liballoc/tests/lib.rs
* /rust/src/liballoc/tests/fmt.rs
* /rust/src/liballoc/tests/vec_deque.rs
* /rust/src/liballoc/tests/slice.rs

This is likely to break cargo build or cargo test as these files may not be
ready to be compiled as a test target today. You can future-proof yourself
and disable this warning by adding `autotests = false` to your [package]
section. You may also move the files to a location where Cargo would not
automatically infer them to be a target, such as in subfolders.

For more information on this warning you can consult
https://github.com/rust-lang/cargo/issues/5330
```
cargo issue: https://github.com/rust-lang/cargo/issues/5330
